bin/utils/options.ml

```diff
@@ -0,0 +1,53 @@
+open Cmdliner
+
+module File = struct
+  let parse_fpath str test_f =
+    let file = Fpath.v str in
+    match test_f file with
+    | Ok true -> `Ok file
+    | Ok false -> `Error (str ^ " is not a file")
+    | Error (`Msg err) -> `Error err
+
+  let fpath = ((fun str -> `Ok (Fpath.v str)), Fpath.pp)
+  let valid_fpath = ((fun str -> parse_fpath str Bos.OS.Path.exists), Fpath.pp)
+  let non_dir_fpath = ((fun str -> parse_fpath str Bos.OS.File.exists), Fpath.pp)
+  let dir_fpath = ((fun str -> parse_fpath str Bos.OS.Dir.exists), Fpath.pp)
+
+  let input =
+    let docv = "FILE" in
+    let doc = "Name of the input file." in
+    Arg.(required & pos 0 (some non_dir_fpath) None & info [] ~doc ~docv)
+
+  let inputs =
+    let docv = "FILE/DIR" in
+    let doc = "Name of the input file or input directory." in
+    Arg.(required & pos 0 (some valid_fpath) None & info [] ~doc ~docv)
+
+  let output =
+    let docv = "FILE" in
+    let doc = "Name of the output file." in
+    Arg.(value & opt (some fpath) None & info [ "o"; "output" ] ~doc ~docv)
+end
+
+let mode =
+  let docv = "MODE" in
+  let doc =
+    "Mode of the execution. Options include: (1) 'c' for Concrete; (2) 'saf' \
+     for Symbolic with Array Fork Memory; (3) 'saite' for Symbolic with Array \
+     ITE Memory; (4) 'st' for Symbolic with Tree Memory; and (5) 'sopl' for \
+     Symbolic with Operation List Memory."
+  in
+  Arg.(required & pos 1 (some string) None & info [] ~doc ~docv)
```

Comment:
Here we would need to create an enumeration type for `type mode` so that we can use it with cmdliner. We can do this with:

```ocaml
let mode_conv =
  Cmdliner.Arg.enum
    [ ("concrete", Cmd_execute.Concrete)
    ; ("saf", Cmd_execute.Saf)
    ; ...
    ]
```

then we change the argument type to take in this required mode:

```ocaml
    Arg.(required & pos 1 (some mode_conv) None ...
```

You can see a better example I did [here](https://github.com/formalsec/encoding/blob/2792eaebfce8857cdad39a4b4887cdf2732fe623/bin/main.ml#L17). Just search for `prover_conv`